client/js/main.js

```diff
@@ -14,22 +14,19 @@ let maxZ = {};
 export const dropTargets = new Map();
 
 function compareDropTarget(widget, t, exclude){
-  const dropTarget = asArray(t.get('dropTarget'))
-  let isValid = false;
-  for(let i = 0; i < dropTarget.length; i++) {
+  for(const dropTargetObject in asArray(t.get('dropTarget'))) {
```

Comment:
Will there be a problem if  dropTarget of t is null?

Reply:
Probably.

Reply:
`t` either comes from the function `getValidDropTargets` or from function `children.filter` in holder.js.  I would think it would always be a holder.

Reply:
A game designer could still set dropTarget to something other than an object or array of objects which could cause problems.